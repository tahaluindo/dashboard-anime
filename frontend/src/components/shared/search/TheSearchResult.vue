<template>
  <p class="font-opensans clamptext-08-12 text-primary italic" v-if="search">
    Exibindo resultado para: <strong>{{ search }}</strong>
  </p>
</template>

<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  props: {
    search: {
      type: String,
    },
  },
});
</script>

<style lang="postcss"></style>
