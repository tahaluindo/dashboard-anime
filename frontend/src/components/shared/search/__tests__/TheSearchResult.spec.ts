import { mount } from "@vue/test-utils";
import TheSearchResult from "../TheSearchResult.vue";

describe("TheSearchResult component test.", () => {
  it("should render the default component.", async () => {
    const wrapper = mount(TheSearchResult, {
      props: {
        search: "teste",
      },
    });

    const pElement = wrapper.find("p").element;

    expect(pElement.innerHTML).toContain("teste");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
