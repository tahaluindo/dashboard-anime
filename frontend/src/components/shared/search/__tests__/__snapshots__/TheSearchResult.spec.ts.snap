// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheSearchResult component test. should render the default component. 1`] = `"<p class=\\"font-opensans clamptext-08-12 text-primary italic mt-3\\"> Exibindo resultado da pesquisa para: <strong>teste</strong></p>"`;
